CONTRIBUTING.md

```diff
@@ -0,0 +1,174 @@
+
+Besides TVM binaries, the Tact compiler generates TypeScript "wrappers" to conveniently test Tact contracts locally in a simulated blockchain environment using, for instance, the standard de-facto [Sandbox](https://github.com/ton-org/sandbox) package.
+
+Additionally, it generates summaries for humans and machines in Markdown and JSON formats.
+The summaries include information like
+
+- binary code size,
+- [TL-B](https://docs.ton.org/develop/data-formats/tl-b-language) schemas for the program types including contract storage and message formats,
+- TVM [exit codes](https://docs.ton.org/learn/tvm-instructions/tvm-exit-codes),
+- trait inheritance and contract dependency diagrams.
+
+Currently, Tact does not have a (formal) language specification, so one needs to consult the [Tact docs](https://docs.tact-lang.org) and the tests in this repository.
+
+The list of known bugs can be obtained using the following GitHub request: <https://github.com/tact-lang/tact/issues?q=is%3Aopen+is%3Aissue+label%3Abug>.
+
+More detail about different part of the compiler including their corresponding entry points can be found below.
+
+## The Tact dependencies
+
+Tact is implemented in TypeScript. The minimum required version for Node.js is 22.
+
+The rest of the build and development dependencies are specified, as usual, in the [package.json](./package.json) file and the most important ones are described in the present document.
+
+Tact's pipeline uses a patched version of the FunC compiler vendored as a WASM binary with some JS wrappers, see the following files:
+
+- [src/func/funcfiftlib.js](./src/func/funcfiftlib.js)
+- [src/func/funcfiftlib.wasm](./src/func/funcfiftlib.wasm)
+- [src/func/funcfiftlib.wasm.js](./src/func/funcfiftlib.wasm.js)
+
+The message of the commit [`f777da3213e3b064a7f407b2569cfd546cca277e`](https://github.com/tact-lang/tact/commit/f777da3213e3b064a7f407b2569cfd546cca277e) explains how the patched version was obtained. We had to patch the FunC compiler because the corresponding [FunC compiler issue](https://github.com/ton-blockchain/ton/issues/971) is unresolved at the time of writing.
+
+## Building Tact
+
+The most up-to-date recipe to build Tact is described in [.github/workflows/tact.yml](./.github/workflows/tact.yml) GitHub Actions file.
+
+## Testing Tact implementation
+
+As our testing framework we use [Jest](https://jestjs.io). We use a combination of snapshot and expectation tests supported by Jest.
+
+Some tests are put in the same directory with the implementation and can be located in `*.spec.ts` files,
+other tests are grouped into categories in the [src/test](./src/test) folder. The project map section has more information on tests relevant for each compiler component.
+
+### How to update test snapshots
+
+Updating all the test snapshots:
+
+```shell
+yarn test --u
+```
+
+Updating a subset of the test snapshots can be done like so:
+
+```shell
+yarn test -u -t='spec-name-pattern'
```

Comment:
you can also just pass one or multiple spec names (or prefixes) in the command like `yarn test -u grammar resolve`

Reply:
Good point too, not sure if we should add more stuff on Jest here, I'd prefer to have some basics

Reply:
> Good point too, not sure if we should add more stuff on Jest here, I'd prefer to have some basics

I mentioned it because it’s a little bit easier than passing it in `-t=""` parameter

Reply:
ah, got it, fixed